### DISCUSSION TITLE
self-hosted instance: email service

### DISCUSSION THREAD

**Discussion by davidsean** [biigle/core#813] [Q&A]
Hello,
I am noticing emails were not being sent (password-reset/analysis reports). Digging around the mail settings in the `.env` file suggested that commenting out `MAIL_DRIVER` would default to the PHP's mail(). I think that would work out nicely for me since I probably need to send only a few emails a month (and I assume my institute won't want me to use an external email service).
That config leads to:  `production.ERROR: Connection could not be established with host "smtp.mailgun.org:587"`

Which makes sense after looking into `config/mail.php`,  as `'smpt'` is the default driver (and mailgun:587 are the default host:port for smpt). 

From the list of drivers, I thought maybe `sendmail` would work, I try `MAIL_DRIVER="sendmail"`, which unfortunately, leads to a 500 Internal Server Error.
`production.ERROR: Unable to write bytes on the wire. {"exception":"[object] (Symfony\\Component\\Mailer\\Exception\\TransportException(code: 0): Unable to write bytes on the wire. at /var/www/vendor/symfony/mailer/Transport/Smtp/Stream/AbstractStream.php:50)`

Suspecting the option is deprecated, I tried interacting with `sendmail`
```bash
docker compose exec -it app /usr/sbin/sendmail
``` 
only to get: `sendmail: can't connect to remote host (127.0.0.1): Connection refused`
So `/usr/sbin/sendmail` is indeed installed in the `app` container, but perhaps there needs to be a mail daemon running? At this point, not knowing much about sendmail, not understanding how its supposed to interact with Biigle, and not having having bash to play in container, I decided to post a question here. :)
Thanks for any suggestions!

**Comment by mzur** [comment]
Hi, it's true that the `mail` driver is no longer available. It must have been dropped in one of the Laravel upgrades. I've now changed the default value to `smtp`. Unfortunately I'm not familiar with `sendmail`, as I've always worked with the `smtp` driver.  You can experiment in the container by running `docker compose exec -it app sh`, which will give you an interactive command line (Bash is not installed in this container). However, many emails are also sent from the worker container, which in this case must also support `sendmail`. From a quick test, this is not the case.

The easiest option may be to use `smtp`. Maybe your IT department can create new SMTP credentials for you internal BIIGLE instance? For testing you could also configure it with your own SMTP credentials.
